modules/comment/comment.service.js

```diff
@@ -50,26 +50,26 @@ class CommentsService {
     const product = await Product.findById(id);
     const { userRates } = product;
     let { rateCount } = product;
-    const { rate } = userRates.find(rate => String(rate.user) === String(user._id)) || {};
+    const { rate } =      userRates.find(rate => String(rate.user) === String(user._id)) || {};
 
-    const rateSum = product.rate * rateCount - (rate ? rate : !!rate) + data.rate;
-    rateCount = rate ? rateCount : ++rateCount
+    const rateSum = product.rate * rateCount - (rate || !!rate) + data.rate;
+    rateCount = rate ? rateCount : ++rateCount;
     const newRate = rateSum / rateCount;
 
     const newUserRates = rate
-        ? userRates.map(item => (String(item.user) === String(user._id)
-          ? { user: item.user, rate: data.rate }
-          : item))
-        : [...userRates, { ...data, user: user._id }]
+      ? userRates.map(item => (String(item.user) === String(user._id)
+        ? { user: item.user, rate: data.rate }
+        : item))
+      : [...userRates, { ...data, user: user._id }];
```

Comment:
It's better to replace this nested ternary operator with `if` statement so it'll become more readable this way